sensorprocess/testhelper.go

```diff
@@ -144,104 +183,172 @@ func onlineModeIMUTestHelper(
 			sensorName:     sensorName,
 			currentReading: currentReading,
 		}
-		calls = append(calls, args)
-		if len(calls) == 1 {
+		imuCalls = append(imuCalls, args)
+		if len(imuCalls) == 1 {
 			return errUnknown
 		}
-		if len(calls) == 2 {
+		if len(imuCalls) == 2 {
+			return cartofacade.ErrUnableToAcquireLock
+		}
+		return nil
+	}
+
+	var odometerCalls []addOdometerReadingArgs
+	cf.AddOdometerReadingFunc = func(
+		ctx context.Context,
+		timeout time.Duration,
+		sensorName string,
+		currentReading s.TimedOdometerReadingResponse,
+	) error {
+		args := addOdometerReadingArgs{
+			timeout:        timeout,
+			sensorName:     sensorName,
+			currentReading: currentReading,
+		}
+		odometerCalls = append(odometerCalls, args)
+		if len(odometerCalls) == 1 {
+			return errUnknown
+		}
+		if len(odometerCalls) == 2 {
 			return cartofacade.ErrUnableToAcquireLock
 		}
 		return nil
 	}
 
 	config.CartoFacade = &cf
-	config.IMU = imu
+	config.MovementSensor = movementSensor
 	config.IsOnline = true
 
-	err = config.addIMUReadingInOnline(ctx)
+	testNumberCalls := func(movementSensor s.TimedMovementSensor, expectedNumberCalls int) {
+		if movementSensor.Properties().IMUSupported {
+			test.That(t, len(imuCalls), test.ShouldEqual, expectedNumberCalls)
+		} else {
+			test.That(t, len(imuCalls), test.ShouldEqual, 0)
+		}
+		if movementSensor.Properties().OdometerSupported {
+			test.That(t, len(odometerCalls), test.ShouldEqual, expectedNumberCalls)
+		} else {
+			test.That(t, len(odometerCalls), test.ShouldEqual, 0)
+		}
+	}
+
+	err = config.addMovementSensorReadingInOnline(ctx)
 	test.That(t, err, test.ShouldBeNil)
-	test.That(t, len(calls), test.ShouldEqual, 1)
+	testNumberCalls(movementSensor, 1)
 
-	err = config.addIMUReadingInOnline(ctx)
+	err = config.addMovementSensorReadingInOnline(ctx)
 	test.That(t, err, test.ShouldBeNil)
-	test.That(t, len(calls), test.ShouldEqual, 2)
+	testNumberCalls(movementSensor, 2)
 
-	err = config.addIMUReadingInOnline(ctx)
+	err = config.addMovementSensorReadingInOnline(ctx)
 	test.That(t, err, test.ShouldBeNil)
-	test.That(t, len(calls), test.ShouldEqual, 3)
+	testNumberCalls(movementSensor, 3)
 
-	for i, call := range calls {
-		t.Logf("call %d", i)
-		test.That(t, call.sensorName, test.ShouldResemble, string(testImu))
-		// the IMU test fixture happens to always return the same readings currently
-		// in reality they are likely different every time
-		test.That(t, call.currentReading.LinearAcceleration, test.ShouldResemble, expectedIMUReading.LinearAcceleration)
-		test.That(t, call.currentReading.AngularVelocity, test.ShouldResemble, expectedIMUReading.AngularVelocity)
-		test.That(t, call.timeout, test.ShouldEqual, config.Timeout)
+	if movementSensor.Properties().IMUSupported {
+		for i, call := range imuCalls {
+			t.Logf("call %d", i)
+			test.That(t, call.sensorName, test.ShouldResemble, string(testMovementSensor))
+			// the IMU test fixture happens to always return the same readings currently
```

Comment:
```suggestion
			// the IMU test fixture happens to always return the same readings currently.
```

[nit] minor but i read it first as ... readings currently in reality which confused me